src/utils/exception.hpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef EXCEPTION_HPP_
+#define EXCEPTION_HPP_
```

Comment:
header files should be separated from source files in some way.
I see two options: 
1) make include / src folders for each subcomponent & make include directories as public properties for targets that depend on them
2) follow core_lib_cpp approach: have root/src and root/include dir.